### PR TITLE
fix: core layout improve alignment slot support

### PR DESCRIPTION
**Pull request by coryrylan**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Certain cds-layout alignments were not possible with slotted elements.

Issue Number: https://github.com/vmware/clarity/issues/5680

## What is the new behavior?
This adds cds-layout alignment support for slotted elements. cds-layouts cant handle all layout options that cross between light and shadow dom however alignments can be supported. This PR also adds a simple example story use case of aligning child items. 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by hippee-lee — APPROVED** [review]

File: packages/core/docs/utilities.stories.ts [hunk 1]
```diff
@@ -49,14 +50,17 @@ export const internationalization = () => {
 export const lazyLoading = () => {
   class LazyDemo extends LitElement {
     @property({ type: Boolean })
-    loaded = false;
+    loaded = true;
```
Comment (mathisscott) [inline packages/core/docs/utilities.stories.ts]: I guess I'm missing why this was changed. It seems now that `loaded` is always true. So don't we lose the lazyLoading aspect of this story now?

File: packages/core/src/styles/layout/_shadow-dom.scss [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  </div>
+
+  <my-element>
+    <button></button>
+    <button></button>
+    <button cds-layout="align:bottom"></button>
+  </my-element>
+*/
+
+@mixin generateShadowDomVerticalStretch($breakpoint: null) {
```
Comment (mathisscott) [inline packages/core/src/styles/layout/_shadow-dom.scss:22]: I like that all of these have been isolated into their own file. 👍

**Review by mathisscott — CHANGES_REQUESTED** [review]

File: packages/core/docs/utilities.stories.ts (hunk 1 above)
Comment (coryrylan) [reply packages/core/docs/utilities.stories.ts]: ah good catch, the dropdown file changes shouldnt be in the PR. I was using it for a test

Comment (coryrylan) [reply packages/core/docs/utilities.stories.ts]: Fixed, sorry for the force push. I messed up my rebase earlier. That file should be removed now.

**Review by mathisscott — APPROVED** [review]

**Pull request merged**